src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -24,8 +24,19 @@ internal sealed class EnumConverter<T> : JsonConverter<T>
 
         private readonly JsonNamingPolicy? _namingPolicy;
 
-        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCache;
-
+        /// <summary>
+        /// Holds a mapping from enum value to text that might be formatted with <see cref="_namingPolicy" />.
+        /// </summary>
+        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _namingPolicyCache;
+
+        /// <summary>
+        /// Holds a mapping from text that might be formatted with <see cref="_namingPolicy" /> to enum value.
+        /// </summary>
+        private ConcurrentDictionary<string, T>? _enumValueCache;
+
+        /// <summary>
+        /// Holds a mapping from enum value to text that might be formatted with a dictionary key policy.
+        /// </summary>
```

Comment:
Technically, a converter can be passed multiple `JsonSerializerOptions` instances with different `DictionaryKeyPolicy`, so I think this cache might actually be invalid. We should probably just remove it.

Reply:
Alternatively, you might store the `JsonSerializerOptions` passed by the factory in a field and have the cache kick in if the passed options matches the stored instance (which should account for 99% of use cases). It would still need to check if the options instance is mutable though.